### DISCUSSION TITLE
Indeterminate attr binding for inputs

### DISCUSSION THREAD

**Discussion by Tylerian** [alpinejs/alpine#2490] [3. Feature ideas]
# Context
Setting indeterminate state on inputs is cumbersome and a whole PITA.

# Problem

For setting the indeterminate state on inputs, we must first get a reference to the input and then manually set it. This can get cumbersome when working on a nested alpine component, as we must walk the element hierarchy tree until we find the input so we can mark it as indeterminate.

# Solution
That whole thing could be easily avoided if the `:indeterminate` binding could detect it's being applied on an input element and automatically did it for us, example:

```html
<input :indeterminate="<bool_expr>" />
```

**Comment by SimoTod** [comment]
indeterminate is not an attribute it's a property which is why your example doesn't work. At the moment Alpine doesn't offer a special directive to set attributes but you can use x-effect:
`<input x-effect="$el.indeterminate = bool_expression">`.

**Reply by Tylerian** [reply]
Yep, that's why this is a feature Idea :)
